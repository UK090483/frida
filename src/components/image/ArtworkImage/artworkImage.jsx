import React, { useEffect, useState } from "react"

export default function ArtworkImage({ src, alt, onLoad }) {
  const [hasMounted, setHasMounted] = useState(false)
  const [show, setShow] = useState(false)

  function transformImage(image, option) {
    var imageService = "https://img2.storyblok.com/"
    var path = image.replace("https://a.storyblok.com", "")
    return imageService + option + "/" + path
  }

  const resizeSrc = transformImage(src, "500x0")

  useEffect(() => {
    setHasMounted(true)
  }, [])

  const handleLoad = () => {
    setShow(true)
  }

  return (
    hasMounted && (
      <img
        style={{
          marginBottom: 0,
          opacity: show ? 1 : 0,
          transition: "opacity 1s",
        }}
        alt={alt}
        onLoad={() => {
          onLoad()
          handleLoad()
        }}
        src={resizeSrc}
      ></img>
    )
  )
}
