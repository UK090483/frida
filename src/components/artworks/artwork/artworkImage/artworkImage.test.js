import React from "react"
import renderer from "react-test-renderer"
import ArtworkImage from "./artworkImage"
import { cleanup, fireEvent, render } from "@testing-library/react"
import "jest-styled-components"

describe("ArtworkImage ", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArtworkImage></ArtworkImage>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
