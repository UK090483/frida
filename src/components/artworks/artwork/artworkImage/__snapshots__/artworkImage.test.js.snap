// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkImage  renders correctly 1`] = `""`;
